framework/include/limiters/VenkatakrishnanLimiter.h

```diff
@@ -0,0 +1,140 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Limiter.h"
+#include "MathFVUtils.h"
+
+namespace Moose
+{
+namespace FV
+{
+/**
+ * @brief Implements the Venkatakrishnan limiter for flux limiting.
+ *
+ * The Venkatakrishnan limiter is used to reduce numerical oscillations and enforce monotonicity
+ * in computational fluid dynamics (CFD) and other numerical simulations. This limiter adjusts
+ * the flux limiting ratio based on face centroids and cell centroids, handling different gradient
+ * conditions.
+ *
+ * The limiter function is derived from the following equations:
+ *
+ * 1. Calculation of the face delta:
+ * \f[
+ * \Delta_{\text{face}} = \nabla \phi_{\text{upwind}} \cdot (\mathbf{x}_{\text{face}} -
+ * \mathbf{x}_{\text{cell}}) \f]
+ *
+ * 2. Calculation of the deltas for maximum and minimum values relative to the upwind value with a
+ * small perturbation to avoid division by zero: \f[ \Delta_{\text{max}} = \phi_{\text{max}} -
+ * \phi_{\text{upwind}} + 1e-10 \f] \f[ \Delta_{\text{min}} = \phi_{\text{min}} -
+ * \phi_{\text{upwind}} + 1e-10 \f]
+ *
+ * 3. Calculation of the gradient ratio coefficient \( r_f \):
+ * \f[
+ * r_f = \begin{cases}
+ * \frac{\Delta_{\text{face}}}{\Delta_{\text{max}}} & \text{if } \Delta_{\text{face}} \geq 0 \\
+ * \frac{\Delta_{\text{face}}}{\Delta_{\text{min}}} & \text{otherwise}
+ * \end{cases}
+ * \f]
+ *
+ * 4. Venkatakrishnan limiter formula:
+ * \f[
+ * \beta(r_f) = \frac{2 r_f + 1.0}{r_f (2 r_f + 1.0) + 1.0}
```

Comment:
do you have a reference for this formula? 
I m seeing a coefficient K in the Ventakrishnan limiter on random references

Reply:
Yes. It is the one that I put in markdown file. Adding here for completness. As per Sewby's diagram, the K factor different than one deteriorates boundness